### ISSUE TITLE
Problem with MorphToMany relationship create action.

### ISSUE THREAD

**Issue by mxm1070** [filamentphp/filament#2896]
### Package

filament/filament

### Package Version

v2.13.9

### Laravel Version

v9.18.0

### Livewire Version

v2.10.5

### PHP Version

8.1.5

### Bug description

I have a MorphToMany relationship I'm switching over to the new relation manger system. 

```php
public function tasks(): MorphToMany
{
    return $this
        ->morphToMany(Task::class, 'subject', 'task_subjects')
        ->withTimestamps();
}
```

However on the create action it throws an error: 

`Call to a member function update() on null`

on Filament\Tables\Actions\CreateAction: 71

```php
if (count($pivotColumns)) {
    $record->{$relationship->getPivotAccessor()}->update($pivotData);
}
```
When I dump these variables right before `ray($data, $pivotColumns, $pivotData));` I can see that `$pivotData` is empty even though `$pivotColumns` contains my timestamp columns.

```
array:4 [▼
  "name" => "1234"
  "description" => null
  "due_at" => null
  "owner_id" => 1
]

array:2 [▼
  0 => "created_at"
  1 => "updated_at"
]

[]
```

If I remove the update line the operation works as intended. I'm not sure if I'm missing something on my end but the simple fix appears to be changing the if statement to either `if (count($pivotColumns) && $pivotData)` or just `if ($pivotData)`

### Relevant log output

```shell
[2022-06-23 01:49:41] local.ERROR: Call to a member function update() on null {"userId":1,"exception":"[object] (Error(code: 0): Call to a member function update() on null at /Users/mmyers/Sites/dealer/vendor/filament/tables/src/Actions/CreateAction.php:71)
[stacktrace]
#0 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Filament\\Tables\\Actions\\CreateAction->Filament\\Tables\\Actions\\{closure}(Array, Object(Filament\\Tables\\Actions\\CreateAction), Object(App\\Filament\\Tenant\\Resources\\TaskResource\\RelationManagers\\TasksRelationManager), NULL)
#1 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#2 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(81): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#3 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Object(Closure), Object(Closure))
#4 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/Container.php(651): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Object(Closure), Array, NULL)
#5 /Users/mmyers/Sites/dealer/vendor/filament/support/src/Concerns/EvaluatesClosures.php(21): Illuminate\\Container\\Container->call(Object(Closure), Array)
#6 /Users/mmyers/Sites/dealer/vendor/filament/support/src/Actions/Concerns/CanCustomizeProcess.php(20): Filament\\Support\\Components\\ViewComponent->evaluate(Object(Closure), Array)
#7 /Users/mmyers/Sites/dealer/vendor/filament/tables/src/Actions/CreateAction.php(76): Filament\\Tables\\Actions\\CreateAction->process(Object(Closure))
#8 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Filament\\Tables\\Actions\\CreateAction->Filament\\Tables\\Actions\\{closure}(Array, Object(Filament\\Forms\\ComponentContainer), Object(App\\Filament\\Tenant\\Resources\\TaskResource\\RelationManagers\\TasksRelationManager), Object(Filament\\Tables\\Actions\\CreateAction), Array, NULL)
#9 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#10 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(81): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#11 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Object(Closure), Object(Closure))
#12 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/Container.php(651): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Object(Closure), Array, NULL)
#13 /Users/mmyers/Sites/dealer/vendor/filament/support/src/Concerns/EvaluatesClosures.php(21): Illuminate\\Container\\Container->call(Object(Closure), Array)
#14 /Users/mmyers/Sites/dealer/vendor/filament/support/src/Actions/Action.php(21): Filament\\Support\\Components\\ViewComponent->evaluate(Object(Closure), Array)
#15 /Users/mmyers/Sites/dealer/vendor/filament/tables/src/Concerns/HasActions.php(83): Filament\\Support\\Actions\\Action->call(Array)
#16 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Filament\\Resources\\RelationManagers\\RelationManager->callMountedTableAction(NULL)
#17 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#18 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#19 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#20 /Users/mmyers/Sites/dealer/vendor/livewire/livewire/src/ComponentConcerns/HandlesActions.php(149): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array)
#21 /Users/mmyers/Sites/dealer/vendor/livewire/livewire/src/HydrationMiddleware/PerformActionCalls.php(38): Livewire\\Component->callMethod('callMountedTabl...', Array, Object(Closure))
#22 /Users/mmyers/Sites/dealer/vendor/livewire/livewire/src/LifecycleManager.php(89): Livewire\\HydrationMiddleware\\PerformActionCalls::hydrate(Object(App\\Filament\\Tenant\\Resources\\TaskResource\\RelationManagers\\TasksRelationManager), Object(Livewire\\Request))
#23 /Users/mmyers/Sites/dealer/vendor/livewire/livewire/src/Connection/ConnectionHandler.php(13): Livewire\\LifecycleManager->hydrate()
#24 /Users/mmyers/Sites/dealer/vendor/livewire/livewire/src/Controllers/HttpConnectionHandler.php(20): Livewire\\Connection\\ConnectionHandler->handle(Array)
#25 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(48): Livewire\\Controllers\\HttpConnectionHandler->__invoke('filament.tenant...')
#26 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Routing/Route.php(261): Illuminate\\Routing\\ControllerDispatcher->dispatch(Object(Illuminate\\Routing\\Route), Object(Livewire\\Controllers\\HttpConnectionHandler), '__invoke')
#27 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Routing/Route.php(204): Illuminate\\Routing\\Route->runController()
#28 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Routing/Router.php(725): Illuminate\\Routing\\Route->run()
#29 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#30 /Users/mmyers/Sites/dealer/vendor/inertiajs/inertia-laravel/src/Middleware.php(88): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#31 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Inertia\\Middleware->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#32 /Users/mmyers/Sites/dealer/vendor/laravel/jetstream/src/Http/Middleware/ShareInertiaData.php(62): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#33 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Laravel\\Jetstream\\Http\\Middleware\\ShareInertiaData->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#34 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#35 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#36 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#37 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#38 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php(59): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#39 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Session\\Middleware\\AuthenticateSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#40 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#41 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#42 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#43 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest(Object(Illuminate\\Http\\Request), Object(Illuminate\\Session\\Store), Object(Closure))
#44 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Session\\Middleware\\StartSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#45 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#46 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#47 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#48 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#49 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#50 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Routing/Router.php(726): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#51 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Routing/Router.php(703): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
#52 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Routing/Router.php(667): Illuminate\\Routing\\Router->runRoute(Object(Illuminate\\Http\\Request), Object(Illuminate\\Routing\\Route))
#53 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Routing/Router.php(656): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
#54 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(167): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
#55 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
#56 /Users/mmyers/Sites/dealer/vendor/livewire/livewire/src/DisableBrowserCache.php(19): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#57 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Livewire\\DisableBrowserCache->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#58 /Users/mmyers/Sites/dealer/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#59 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Barryvdh\\Debugbar\\Middleware\\InjectDebugbar->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#60 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#61 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#62 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(36): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#63 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#64 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#65 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#66 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#67 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#68 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Http/Middleware/HandleCors.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#69 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Http\\Middleware\\HandleCors->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#70 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#71 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Http\\Middleware\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#72 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#73 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(142): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#74 /Users/mmyers/Sites/dealer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(111): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#75 /Users/mmyers/Sites/dealer/public/index.php(52): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#76 /Users/mmyers/.composer/vendor/laravel/valet/server.php(234): require('/Users/mmyers/S...')
#77 {main}
"}
```

**Comment by sten** [comment]
I am having the same error in a RelationManager with the CreateAction on a BelongsToMany relationship using `withTimestamps()` on the relationship.

The problem is in this piece of code on line 67 of CreateAction:

```
if (count($pivotColumns)) {
    $record->{$relationship->getPivotAccessor()}->update($pivotData);
}
```

It resolves to `$record->pivot` which returns null. I think the pivot var is not set here, because we are using the newly created record. In examples I see it being used when iterating over the records of a relationships, and when I query it, I get the pivot var.

I think this could be fixed by using the `$joining` attribute to pass the `$pivotData` in the create function of the `$relationship` on line 61.
```
create(array $attributes = [], array $joining = [], $touch = true)
```

I tried it and the code from line 49 onwards, looks like this:

```
$record = $this->process(function (array $data) use ($model): Model {
                $relationship = $this->getRelationship();

                if (! $relationship) {
                    return $model::create($data);
                }

                if ($relationship instanceof BelongsToMany) {
                    $pivotColumns = $relationship->getPivotColumns();
                    $data = Arr::except($data, $pivotColumns);
                }

                $pivotData = [];
                if ($relationship instanceof BelongsToMany) {
                    $pivotData = Arr::only($data, $pivotColumns);
                }

                $record = $relationship->create($data, $pivotData);

                return $record;
            });
```

@danharrin Let me know if you want a PR for this.

**Comment by danharrin** [comment]
All PRs for issues are much appreciated, and are prioritised for review over new feature PRs

**Comment by danharrin** [comment]
Fixed by #3041.
